bundles/EcommerceFrameworkBundle/EventListener/Frontend/TrackingCodeFlashMessageListener.php

```diff
@@ -65,16 +68,40 @@ public function onKernelRequest(GetResponseEvent $event)
             return;
         }
 
-        $trackedCodes = $this->session->getFlashBag()->get(self::FLASH_MESSAGE_BAG_KEY);
+        // Check FlashBag cookie exists to avoid autostart session by accessing the FlashBag.
+        $flashBagCookie = (bool)$request->cookies->get(self::FLASH_MESSAGE_BAG_KEY, false);
+        if ($flashBagCookie) {
+            $trackedCodes = $this->session->getFlashBag()->get(self::FLASH_MESSAGE_BAG_KEY);
 
-        if (is_array($trackedCodes) && count($trackedCodes)) {
-            foreach ($this->trackingManger->getTrackers() as $tracker) {
-                if ($tracker instanceof TrackingCodeAwareInterface && isset($trackedCodes[get_class($tracker)])) {
-                    foreach ($trackedCodes[get_class($tracker)] as $trackedCode) {
-                        $tracker->trackCode($trackedCode);
+            if (is_array($trackedCodes) && sizeof($trackedCodes)) {
```

Comment:
Please, don't use this alias and stick to `count` for consistency reasons.

Reply:
unrelated....thanks for the catch!